Math/src/main/java/org/tribuo/math/neighbor/NeighborsBruteForce.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.math.neighbor;
+
+import com.oracle.labs.mlrg.olcut.util.Pair;
+import org.tribuo.math.distance.DistanceType;
+import org.tribuo.math.la.SGDVector;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A brute-force nearest neighbor query implementation.
+ */
+public class NeighborsBruteForce implements NeighborsQuery{
+
+    private final SGDVector[] data;
+    private DistanceType distanceType = DistanceType.EUCLIDEAN;
+    private int numThreads = 1;
+
+
+    /**
+     * Constructs a brute-force nearest neighbor query object using the provided data.
+     * @param data the data that will be used for meighbor queries.
+     */
+    public NeighborsBruteForce(SGDVector[] data) {
+        this.data = data;
+    }
+
+    /**
+     * Constructs a brute-force nearest neighbor query object using the supplied parameters.
+     * @param data the data that will be used for neighbor queries.
+     * @param distanceType The distance function.
+     */
+    public NeighborsBruteForce(SGDVector[] data, DistanceType distanceType) {
+        this(data);
+        this.distanceType = distanceType;
+    }
+
+    /**
+     * Constructs a brute-force nearest neighbor query object using the supplied parameters.
+     * @param data the data that will be used for neighbor queries.
+     * @param distanceType The distance function.
+     * @param numThreads The number of threads to be used to parallelize the computation.
+     */
+    public NeighborsBruteForce(SGDVector[] data, DistanceType distanceType, int numThreads) {
+        this(data, distanceType);
+        this.numThreads = numThreads;
+    }
+
+    @Override
+    public List<Pair<Integer, Double>> query(SGDVector point, int k) {
```

Comment:
This logic is mostly duplicated from the `calculateCoreDistances` method in `HdbscanTrainer`.

Reply:
The implementation in `KNNModel.innerPredict` is definitely a more optimal way to implement the brute-force method. Thanks for pointing this out.